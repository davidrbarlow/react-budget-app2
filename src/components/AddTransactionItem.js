import React from 'react';
import {connect} from 'react-redux';
import { Route , withRouter} from 'react-router-dom';
import TransactionItemForm from './TransactionItemForm';
import {startAddTransaction}  from '../actions/transactions';
import {toggleAddTransaction}  from '../actions/pageEdits';

export class AddTransactionItem extends React.Component {
 
  constructor(props) {
    super(props);
  
  }

    onSubmit = async (transaction) => {
      console.log('addtransitem', this.props);
        await this.props.startAddTransaction(transaction);
        // window.location.reload();
        await this.props.handleState();
        console.log('props in on submit ',this.props);
        this.props.toggleAddTransaction();
    };

    render() {
        return(
        <div>
            <TransactionItemForm 
            onSubmit={this.onSubmit}
            />
        </div>
    );
    }

}


const mapDispatchToProps = (dispatch) => ({
  startAddTransaction: (transaction) => dispatch(startAddTransaction(transaction)),
  toggleAddTransaction: () => dispatch(toggleAddTransaction())
})

export default connect(undefined, mapDispatchToProps)(withRouter(AddTransactionItem));