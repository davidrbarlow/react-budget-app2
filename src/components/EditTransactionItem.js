import React from 'react';
import {connect} from 'react-redux';
import TransactionItemForm from './TransactionItemForm';

const EditTransactionItem = (props) => {

  return (
    <div>
      <TransactionItemForms
        transaction = {props.transaction}
        onSubmit={(transaction)=>{
        props.dispatch(startEditTransaction(props.transaction.id, transaction));
        console.log(props.match.params.id);
      }}
      />


    </div>
  )
}

const mapStateToProps = (state, props) => {
  return {
      expense: state.transctions.find((transaction)=>transaction.id === props.match.params.id)
  };
};
export default connect(mapStateToProps)(EditExpensePage);