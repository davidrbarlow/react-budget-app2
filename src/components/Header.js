import React from 'react';
import {connect} from 'react-redux';
import {logout} from '../actions/auth.js';


class Header extends React.Component {
    constructor(props) {
        super(props);

        this.handleLogout = this.handleLogout.bind(this);

    }

    handleLogout = () => {
        const token = this.props.authToken;
        console.log('authToken is ',this.props.authToken);
        this.props.logout(token);
    }

    render() {
        return (
            <div className="header">
            Budget App
            <button className="header-button" onClick={this.handleLogout}>Logout</button>
        </div>
        )
    }
}

const mapDispatchToProps = (dispatch) => {
    return{
        logout: (token) => dispatch(logout(token))
    };
} ;

const mapStateToProps = (state) => {
    return {
        authToken: state.auth.authToken,
    }
}


export default connect(mapStateToProps, mapDispatchToProps)(Header);