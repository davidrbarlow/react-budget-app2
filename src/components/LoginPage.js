import React from 'react';
import {connect} from 'react-redux';
import {login} from '../actions/auth';
import {Link} from 'react-router-dom';


class LoginPage extends React.Component {
    constructor(props) {
        super(props);

        this.state = {};

        this.handleSubmit = this.handleSubmit.bind(this);
        this.autoComplete = 'yes';
        this.errorMessage = 'Username or Password is incorrect!'

    };

    onEmailChange = (e) =>{
        const email = e.target.value;
        this.setState(()=>({email}));
    };

    onPasswordChange = (e) =>{
        const password = e.target.value;
        this.setState(()=>({password}));
    };

    handleSubmit = (e) =>{
        e.preventDefault();
        let { email, password } = this.state;
        this.props.login(email, password);
        this.setState({
        email: '',
        password: ''
        });
        document.getElementById("login-form").reset();
  }


render(){
    //let {email, password} = this.state;
    let {isLoginPending, isLoginSuccess, loginError} = this.props;
    return(
        <div className="login-container">
            <div className="login-content-box">
                <div className="login-header">
                    <h1>Budget</h1>
                    <p>It's time to get your expenses under control.</p>
                </div>
                <div className="login">  
                    <form 
                    onSubmit={this.handleSubmit} id="login-form" 
                    autoComplete={this.autoComplete}
                    className="login-form">
                        
                        <div className="">
                        <input
                            type="text"
                            name="email"
                            placeholder="Email"
                            autoFocus={true}
                            className="input-text input-text__login"
                            //value={this.state.emal}
                            onChange={this.onEmailChange}
                        />  
                        </div>
                        <div>
                        <input
                            name="password"
                            type="text"
                            placeholder="Password"
                            className="input-text input-text__login"
                            //value={this.state.password}
                            onChange={this.onPasswordChange}
                        />
                        
                        </div>
                        <div className="login-page-sign-in-up">
                        <button className="button button__submit">
                        Login
                        </button>
                        <div>
                            { isLoginPending && <div>Please wait...</div> }
                            { isLoginSuccess && <div>Success.</div> }
                            { loginError && <div>{this.errorMessage}</div> }
                        </div>
                        <Link to="/signup">Sign up</Link>
                        </div>  
                    </form>
                </div>
            </div>
        </div>
        
        );
    };
};

const mapStateToProps = (state)=>{
    return {
    isLoginPending: state.isLoginPending,
    isLoginSuccess: state.isLoginSuccess,
    loginError: state.loginError,
    authToken: state.authToken
    }
}

const mapDispatchToProps = (dispatch) => {
    return{
        login: (email,password) => dispatch(login(email,password))
    };
};
// const mapDispatchToProps = (dispatch) => ({
//     startLogin2: () => dispatch(startLogin2())
// });



//export default connect(undefined, mapDispatchToProps)(LoginPage);

// cons mapDispatchToProps

export default connect(mapStateToProps, mapDispatchToProps)(LoginPage);

//export default LoginPage;

