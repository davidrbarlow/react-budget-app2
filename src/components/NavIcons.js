import React from 'react';
import {connect} from 'react-redux';
import {Link} from 'react-router-dom';
import {setActivePage} from '../actions/pageEdits';

const NavIcons = (props) => (
    <div className="nav-icons">
      
    <Link to='dashboard'>
      <img id={'transactionsButton'} className="nav-icon"  src={'/images/transactions.svg'} 
      alt={'Transactions'} onClick={()=>{props.setActivePage('dashboard')}}></img>
    </Link>
      
      
    <Link to='/projection'> 
      <img id={'projectionButton'} className="nav-icon"  src={'images/projection.svg'} 
      alt={'Projections'} onClick={()=>{props.setActivePage('projection')}}
      ></img>
    </Link>
      
    </div>
);

const mapDispatchToProps = (dispatch) => ({
  setActivePage: (page) => dispatch(setActivePage(page))
});


export default connect(undefined, mapDispatchToProps)(NavIcons);

