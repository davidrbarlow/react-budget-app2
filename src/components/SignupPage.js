import React from 'react';
import {connect} from 'react-redux';
import {signup} from '../actions/auth';

export class SignupPage extends React.Component {
    constructor(props) {
        super(props);

        this.state = {
            // pwLengthPass: false,
            // pwUpperPass: false,
            // pwNumberPass:false,
            // pwSpecialPass:false,
            // pwMatchPass: false,
            //  emValidPass: true
        };

        this.onPasswordChange = this.onPasswordChange.bind(this);       
    };


    checkPasswordLength = (password) => {
        password.length>=0 && password.length < 8 ? this.setState({pwLengthColor: 'red', pwLengthPass: false}) 
       : this.setState({pwLengthColor: 'green', pwLengthPass:true})  ; 
    }

    checkPasswordUpper= (password) => {
       let upperCharacter = (/[A-Z]/.test(password));
            upperCharacter ? this.setState({pwUpperColor: 'green', pwUpperPass: true}) 
        : this.setState({pwUpperColor: 'red', pwUpperPass:false})  ; 
     }

    checkPasswordNumber= (password) => {
        let number = (/[0-9]/.test(password));
        number ? this.setState({pwNumberColor: 'green', pwNumberPass: true}) 
         : this.setState({pwNumberColor: 'red', pwNumberPass:false})  ; 
      }

    checkPasswordSpecial= (password) => {
      
       const pattern= /[ !@#$%^&*()_+\-=[\]{};':"\\|,.<>?]/;
        let specialCharacter =pattern.test(password);
       specialCharacter ? this.setState({pwSpecialColor: 'green', pwSpecialPass: true}) 
         : this.setState({pwSpecialColor: 'red', pwSpecialPass:false})  ; 
      }

    checkPasswordMatch= (password, password2) => {
        (password2 === password) ? this.setState({pwMatchColor: 'green', pwMatchPass: true}) 
        : this.setState({pwMatchColor: 'red', pwMatchPass: false}) 
        
       }

    checkValidEmail = (email) => {
        var pattern = /^(([^<>()[\]\\.,;:\s@"]+(\.[^<>()[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/;
        let validEmail =pattern.test(email);
        validEmail ? this.setState({emailValidColor: 'green', emailValidPass: true}) 
          : this.setState({emailValidColor: 'red', emailValidPass:false})  ; 
    }

    onEmailChange = (e) =>{
        const email = e.target.value;
        this.setState(()=>({email}));
        this.checkValidEmail(email);
    };

    onPasswordChange = (e) => {
        
        const password = e.target.value;
        const password2 = this.state.password2;
        this.setState(()=>({password}));
        this.checkPasswordLength(password);
        this.checkPasswordUpper(password);
        this.checkPasswordSpecial(password);
        this.checkPasswordNumber(password);
        this.checkPasswordMatch(password, password2);
    }

    onPassword2Change = (e) => {
        const password2 = e.target.value;
        this.setState(()=>({password2}));
        let password = this.state.password;
        this.checkPasswordMatch(password, password2);
    }

    handleSubmit =  (e) => {
        e.preventDefault();
        let { email, password} = this.state;
        this.checkEmailPwPass().then((r)=>{
            this.props.signup(email, password)
            .catch((res)=>{
                console.log('email already in use');
                res ? this.setState({error: 'Email already registered'}) : console.log("search failed");      
            }).catch((e)=>{console.log('no match')});
      })
    };

    checkEmailPwPass = async () => {
        
        if (this.state.pwLengthPass && this.state.pwUpperPass && this.state.pwNumberPass && this.state.pwSpecialPass 
            && this.state.pwMatchPass && this.state.emailValidPass) {  
                console.log('async',this.state) ;
                this.setState({error: ''});
                return 'return value from async';
        } else {
                this.setState({error: 'Please enter valid username/password'});
                throw new Error('no match in promise')}
    } 

    render(){
        return(
        <div>
            <h1>Budget</h1>
            <p>It's time to get your expenses under control.</p>
            {this.state.error && <p>{this.state.error}</p>}
            <form onSubmit={this.handleSubmit} id="login-form" className="signup-form">
            <div className="login-text-input">
                <input
                type="text"
                name="email"
                placeholder="email"
                autoFocus
                className="input-text input-text__login"
                onChange={this.onEmailChange}
                />
                <input
                name="password"
                type="password"
                placeholder="Password"
                className="input-text input-text__login"
                onChange={this.onPasswordChange}
                />
                <input
                name="password2"
                type="password"
                placeholder="Re-enter Password"
                className="input-text input-text__login"
                onChange={this.onPassword2Change}
                />
            </div>     
            <span style={{color:this.state.emailValidColor}}>enter valid email</span>
            <span style={{color:this.state.pwLengthColor}}>password must be 8 characters</span>
            <span style={{color:this.state.pwUpperColor}}>password must contain a capital letter</span>
            <span style={{color:this.state.pwSpecialColor}}>Password must contain special character</span>
            <span style={{color:this.state.pwNumberColor}}>Password must contain a number</span>
            <span style={{color:this.state.pwMatchColor}}>Passwords must match</span>
            <button className="button button__submit">
            Signup
            </button>
            </form>
        </div>
        );
    };

};

const mapDispatchToProps = (dispatch) => {
    return{
        signup: (email,password) => dispatch(signup(email,password))
    };
};


export default connect(undefined, mapDispatchToProps)(SignupPage);