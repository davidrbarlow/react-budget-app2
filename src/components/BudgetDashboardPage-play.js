import React from 'react';
import {connect} from 'react-redux';
import NavIcons from './NavIcons';
import Header from './Header';
import TransactionList from './TransactionList';
import TransactionListFilters from './TransactionListFilters';
import TransactionListControls from './TransactionListControls';

import { logout } from '../actions/auth';


class BudgetDashboardPage extends React.Component {

    constructor(props) {
        super(props);
        this.onUnload = this.onUnload.bind(this); // if you need to bind callback to this
    }

    componentWillUnmount() {
        // Make sure to remove the DOM listener when the component is unmounted.
        this.nv.removeEventListener("nv-enter", this.handleNvEnter);
      }

      componentDidMount() {
        // When the component is mounted, add your DOM listener to the "nv" elem.
        // (The "nv" elem is assigned in the render function.)
        this.nv.addEventListener("nv-enter", this.handleNvEnter);
      }

      handleNvEnter = (event) => {
        console.log("Nv Enter:", event);
      }


    onUnload = (event) => {
        console.log("hellooww");
        event.returnValue = "Hellooww";
        this.props.logout(this.props.token);
    }

    render() {
        return (
            <div className="layout">
                <div className="layout--top-left-corner"></div>
                <div className="layout--header">
                    <Header />
                </div>
                <div className="layout--nav-icon">
                    <NavIcons />
                </div>
                <div className="layout--filters">
                    <TransactionListFilters />
                </div>
            
                <div className="layout--controls">
                    <TransactionListControls />
                </div>
                <div className="layout--transactions">
                    <TransactionList />
                </div>       
            </div>
        );
    }
}

const mapDispatchToProps = (dispatch) => {
    return{
        logout: (token) => dispatch(logout(token))
    };
} ;

const mapStateToProps = (state) => {
    return {
        authToken: state.auth.authToken,
    }
}


export default connect(mapStateToProps, mapDispatchToProps)(BudgetDashboardPage);