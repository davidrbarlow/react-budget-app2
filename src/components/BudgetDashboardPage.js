import React from 'react';
import NavIcons from './NavIcons';
import Header from './Header';
import TransactionList from './TransactionList';
import TransactionListFilters from './TransactionListFilters';
import TransactionListControls from './TransactionListControls';


const BudgetDashboardPage = () => (
    <div className="layout">
        <div className="layout--top-left-corner"></div>
        <div className="layout--header">
            <Header />
        </div>
        <div className="layout--nav-icon">
            <NavIcons />
        </div>
        <div className="layout--filters">
            <TransactionListFilters />
        </div>
       
        <div className="layout--controls">
             <TransactionListControls />
        </div>
        <div className="layout--transactions">
            <TransactionList />
        </div>
        
        
        
    </div>
);

export default BudgetDashboardPage;