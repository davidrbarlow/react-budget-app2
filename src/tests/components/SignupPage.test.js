import React from 'react';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';
import { SignupPage } from '../../components/SignupPage';

let wrapper;

beforeEach(()=> {
  wrapper = shallow(<SignupPage />);
});

test('shoud render Signup Page correctly',()=> { 
  expect(toJSON(wrapper)).toMatchSnapshot();
});

test('shoud set email valid true',()=> { 
  const value = 'abc@123.com';
  wrapper.find('input').at(0).simulate('change', {
    target: {value}
  });
  expect(wrapper.state('emailValidPass')).toBe(true);
});

test('shoud set email valid false',()=> { 
  const value = 'abc@123';
  wrapper.find('input').at(0).simulate('change', {
    target: {value}
  });
  expect(wrapper.state('emailValidPass')).toBe(false);
});

test('shoud set password length, number, upper, special to true',()=> { 
  const value = 'Test123!';
 // console.log(wrapper.find('input').get(1))
  wrapper.find('input').at(1).simulate('change', {
    target: {value}
  });
  expect(wrapper.state('pwLengthPass')).toBe(true);
  expect(wrapper.state('pwUpperPass')).toBe(true);
  expect(wrapper.state('pwNumberPass')).toBe(true);
  expect(wrapper.state('pwSpecialPass')).toBe(true);
});

test('shoud set password length, number, upper, special to false',()=> { 
  const value = 'asdf';
 // console.log(wrapper.find('input').get(1))
  wrapper.find('input').at(1).simulate('change', {
    target: {value}
  });
  expect(wrapper.state('pwLengthPass')).toBe(false);
  expect(wrapper.state('pwUpperPass')).toBe(false);
  expect(wrapper.state('pwNumberPass')).toBe(false);
  expect(wrapper.state('pwSpecialPass')).toBe(false);
});

test('shoud set password match true',()=> { 
  const password = 'Test123!';
  const password2 = 'Test123!';
 // console.log(wrapper.find('input').get(1))
  wrapper.find('input').at(1).simulate('change', {
    target: {value : password}
  });
  wrapper.find('input').at(2).simulate('change', {
    target: {value : password2}
  });
  expect(wrapper.state('pwMatchPass')).toBe(true);
});

test('shoud set password match false',()=> { 
  const password = 'Test123!';
  const password2 = 'Test123';
 // console.log(wrapper.find('input').get(1))
  wrapper.find('input').at(1).simulate('change', {
    target: {value : password}
  });
  wrapper.find('input').at(2).simulate('change', {
    target: {value : password2}
  });
  expect(wrapper.state('pwMatchPass')).toBe(false);
});


test('should call onSubmit prop for valid form submission', () => {
  const checkEmailPwPass = jest.fn(() => (Promise.resolve({signup : () => {}})));
  wrapper.instance().checkEmailPwPass = checkEmailPwPass;
  wrapper.update();
  const state = {email : 'abc123@test.com' , password: 'Test123!'}
  wrapper.find('form').simulate('submit', {
      preventDefault: () => {}
  });
  expect(checkEmailPwPass).toHaveBeenCalled();
});