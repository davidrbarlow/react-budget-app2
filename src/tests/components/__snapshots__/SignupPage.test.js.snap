// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`shoud render Signup Page correctly 1`] = `
<div>
  <h1>
    Budget
  </h1>
  <p>
    It's time to get your expenses under control.
  </p>
  <form
    className="signup-form"
    id="login-form"
    onSubmit={[Function]}
  >
    <div
      className="login-text-input"
    >
      <input
        autoFocus={true}
        className="input-text input-text__login"
        name="email"
        onChange={[Function]}
        placeholder="email"
        type="text"
      />
      <input
        className="input-text input-text__login"
        name="password"
        onChange={[Function]}
        placeholder="Password"
        type="password"
      />
      <input
        className="input-text input-text__login"
        name="password2"
        onChange={[Function]}
        placeholder="Re-enter Password"
        type="password"
      />
    </div>
    <span
      style={
        Object {
          "color": undefined,
        }
      }
    >
      enter valid email
    </span>
    <span
      style={
        Object {
          "color": undefined,
        }
      }
    >
      password must be 8 characters
    </span>
    <span
      style={
        Object {
          "color": undefined,
        }
      }
    >
      password must contain a capital letter
    </span>
    <span
      style={
        Object {
          "color": undefined,
        }
      }
    >
      Password must contain special character
    </span>
    <span
      style={
        Object {
          "color": undefined,
        }
      }
    >
      Password must contain a number
    </span>
    <span
      style={
        Object {
          "color": undefined,
        }
      }
    >
      Passwords must match
    </span>
    <button
      className="button button__submit"
    >
      Signup
    </button>
  </form>
</div>
`;
